frontend/src/pages/ArtifactDetails.tsx

```diff
@@ -88,38 +135,43 @@ export default class ArtifactDetails extends Page<{}, ArtifactDetailsState> {
   }
 
   private async load(): Promise<void> {
-    const getArtifactsRequest = new GetArtifactsByIDRequest();
-    getArtifactsRequest.setArtifactIdsList([this.id]);
-    Apis.getMetadataServiceClient().getArtifactsByID(getArtifactsRequest, (err, res) => {
-      if (err) {
-        this.showPageError(serviceErrorToString(err));
-        return;
-      }
-
-      if (!res || !res.getArtifactsList().length) {
-        this.showPageError(`No ${this.fullTypeName} identified by id: ${this.id}`);
-        return;
-      }
-
-      if (res.getArtifactsList().length > 1) {
-        this.showPageError(`Found multiple artifacts with ID: ${this.id}`);
-        return;
-      }
-
-      const artifact = res.getArtifactsList()[0];
-
-      const artifactName =
-        getResourceProperty(artifact, ArtifactProperties.NAME) ||
-        getResourceProperty(artifact, ArtifactCustomProperties.NAME, true);
-      let title = artifactName ? artifactName.toString() : '';
-      const version = getResourceProperty(artifact, ArtifactProperties.VERSION);
-      if (version) {
-        title += ` (version: ${version})`;
-      }
-      this.props.updateToolbar({
-        pageTitle: title,
-      });
-      this.setState({ artifact });
+    const request = new GetArtifactsByIDRequest();
+    request.setArtifactIdsList([Number(this.id)]);
```

Comment:
Not sure what's a recommended way here, your's might be more readable. But you could also write this more simply as:
```suggestion
    request.setArtifactIdsList([+this.id]);
```

Reply:
The TypeScript style guide prefers Number() for type coercion.